#!/usr/bin/env python
PACKAGE = "delta_manipulator"
 
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("TORQUE_ENABLE",   bool_t,   0, "Torque on/off?",  True)
gen.add("READ_POSITIONS",   bool_t,   0, "Read positions of servos?",  False)
gen.add("CHANGE_PARAMS",   bool_t,   0, "Allow changes to control table params?",  False)
gen.add("POSITION_D_GAIN", int_t, 0, "D gain",    0, 0,   16000)
gen.add("POSITION_I_GAIN", int_t, 0, "I gain",    0, 0,   16000)
gen.add("POSITION_P_GAIN", int_t, 0, "P gain",    930, 0,   2000)
gen.add("FF_ACCEL_GAIN", int_t, 0, "Acceleration Feedforward",    0, 0,  16000)
gen.add("FF_VEL_GAIN", int_t, 0, "Velocity Feedforward",    0, 0,  16000)
gen.add("PROFILE_ACCELERATION", int_t, 0, "Profile acceleration",    0, 0,  16000)
gen.add("PROFILE_VELOCITY", int_t, 0, "Profile velocity",    100, 0,  16000)

exit(gen.generate(PACKAGE, "delta_manipulator", "Servo"))